retrieval/target.go

```diff
@@ -363,6 +366,17 @@ func (t *target) URL() string {
 	return t.url
 }
 
+// PublicURL implements Target.
```

Comment:
If we're only going to attach `<host>:<port>` as an instance label now, the name of this method should probably change to `InstanceLabel` or something like that.